Microsoft.Toolkit.Uwp.UI.Controls/Menu/Menu.Events.cs

```diff
@@ -0,0 +1,306 @@
+            if (_wrapPanel != null)
+            {
+                _wrapPanel.Orientation = Orientation;
+            }
+
+            LostFocus -= Menu_LostFocus;
+            LostFocus += Menu_LostFocus;
+            Dispatcher.AcceleratorKeyActivated -= Dispatcher_AcceleratorKeyActivated;
+            Window.Current.CoreWindow.KeyDown -= CoreWindow_KeyDown;
+            Dispatcher.AcceleratorKeyActivated += Dispatcher_AcceleratorKeyActivated;
+            Window.Current.CoreWindow.KeyDown += CoreWindow_KeyDown;
+        }
+
+        private void ClassicMenu_Unloaded(object sender, RoutedEventArgs e)
+        {
+            Dispatcher.AcceleratorKeyActivated -= Dispatcher_AcceleratorKeyActivated;
+            Window.Current.CoreWindow.KeyDown -= CoreWindow_KeyDown;
+        }
+
+        private void CoreWindow_KeyDown(CoreWindow sender, KeyEventArgs args)
+        {
+            if (IsInTransitionState)
+            {
+                return;
+            }
+
+            var element = FocusManager.GetFocusedElement();
+
+            if (NavigateUsingKeyboard(element, args, this, Orientation))
+            {
+                return;
+            }
+
+            string gestureKey = MapInputToGestureKey(args.VirtualKey);
+
+            if (gestureKey == null)
+            {
+                return;
+            }
+
+            if (MenuItemInputGestureCache.ContainsKey(gestureKey))
+            {
+                var cachedMenuItem = MenuItemInputGestureCache[gestureKey];
+                if (cachedMenuItem is MenuFlyoutItem)
+                {
+                    var menuFlyoutItem = (MenuFlyoutItem)cachedMenuItem;
+                    menuFlyoutItem.Command?.Execute(menuFlyoutItem.CommandParameter);
+                }
+            }
+        }
+
+        private void Menu_LostFocus(object sender, RoutedEventArgs e)
+        {
+            _isLostFocus = true;
+
+            if (AllowTooltip)
+            {
+                HideSubItemTooltips();
+            }
+        }
+
+        private void Dispatcher_AcceleratorKeyActivated(CoreDispatcher sender, AcceleratorKeyEventArgs args)
+        {
+            _lastFocusElement = FocusManager.GetFocusedElement() as Control;
+            if (args.VirtualKey == VirtualKey.Menu && !args.KeyStatus.WasKeyDown)
+            {
+                _altHandled = false;
+            }
+            else if (args.VirtualKey == VirtualKey.Menu && args.KeyStatus.IsKeyReleased && !_altHandled)
+            {
+                if (!IsOpened)
+                {
+                    if (_isLostFocus)
+                    {
+                        LostFocus -= Menu_LostFocus;
+                        Focus(FocusState.Programmatic);
+                        _lastFocusElementBeforeMenu = _lastFocusElement;
+                        _isLostFocus = false;
+
+                        if (AllowTooltip)
+                        {
+                            ShowSubItemToolTips();
```

Comment:
The user will choose between AccessKeys and InputGesture there is no need to check